### PR TITLE
Package documentation and examples

### PR DESCRIPTION
**Pull request by Davejkane**
This pull request creates package documentation and some examples.

### CODE REVIEW

File: doc_test.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
+package trie
+
+import "fmt"
+
+func Example() {
```
Comment (glaslos) [inline doc_test.go:5]: https://pkg.go.dev/github.com/fluhus/godoc-tricks#Examples

```suggestion
func ExampleSearchAll() {
```

File: doc_test.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+	fmt.Println(results2)
+
+	// Output:
+	// [Wednesday]
+	// [Thursday, Tuesday, Wednesday]
+	//
+	// The second result demonstrates both fuzzy matching and the default Levenshtein distance scheme. This is configurable.
+}
+
+func Example_noFeatures() {
```
Comment (glaslos) [inline doc_test.go:22]:
```suggestion
func ExampleSearchAll_noFeatures() {
```

**Review by glaslos — APPROVED** [review]
If you rename your tests to Example{function_name}_{misc_suffix} they will show up in godoc.
Merge at your digression.

**Comment by Davejkane** [comment]
I think the examples are more general than highlighting that particular method. I'm going to leave it as it is for now.

**Pull request merged**